src/extension.ts

```diff
@@ -721,9 +733,9 @@ function buildPushThenExec(fn) {
                         vscode.window.showInformationMessage(image + ' pushed.');
```

Comment:
Before executing `push`, it's better to add user consent `This command needs to push your docker image to docker hub, do you want to continue?`

Reply:
Yeah, this has been raised as #80.  As noted there, this also gets us thinking about giving options for ACR etc., private registries and local registry (for minikube), and aligning with how vscode-docker handles this stuff.  Adding to the complexity, a consent prompt entails tracking that the user has consented so we don't bug them every time.  I'd like to restrict the scope of this PR to giving better errors in the existing failure cases, and tackle the broader issue of our interaction with registries in a subsequent one.

Reply:
Ok to do it on the new PR.